From 2ff2f05bf3b81e817eb0246021cd1b3265dffc2f Mon Sep 17 00:00:00 2001
From: Chao Sun <sunchao@apache.org>
Date: Fri, 10 Mar 2017 08:55:31 -0800
Subject: [PATCH 1039/1164] CDH-51571 HIVE-16175: Possible race condition in
 InstanceCache (Chao Sun, reviewed by Xuefu Zhang)

(cherry picked from commit 1047126a3a91006a7a8395fec539825da43b1261)

Change-Id: I6b690762459acd265dbc266c02f9a521710ee8b2
---
 .../hadoop/hive/serde2/avro/InstanceCache.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/serde/src/java/org/apache/hadoop/hive/serde2/avro/InstanceCache.java b/serde/src/java/org/apache/hadoop/hive/serde2/avro/InstanceCache.java
index c7289cb..3a1a458 100644
--- a/serde/src/java/org/apache/hadoop/hive/serde2/avro/InstanceCache.java
+++ b/serde/src/java/org/apache/hadoop/hive/serde2/avro/InstanceCache.java
@@ -50,7 +50,7 @@ public Instance retrieve(SeedObject hv) throws AvroSerdeException {
    * Retrieve (or create if it doesn't exist) the correct Instance for this
    * SeedObject using 'seenSchemas' to resolve circular references
    */
-  public Instance retrieve(SeedObject hv,
+  public synchronized Instance retrieve(SeedObject hv,
       Set<SeedObject> seenSchemas) throws AvroSerdeException {
     if(LOG.isDebugEnabled()) LOG.debug("Checking for hv: " + hv.toString());
 
-- 
1.7.9.5

